cinema/views.py

```diff
@@ -101,8 +105,25 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        ["POST"],
+        detail=True,
+        url_path="upload-image",
+        permission_classes=(IsAdminUser,),
+    )
+    def upload_image(self, request: Request, pk: int = None) -> Response:
```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]